apps/gateway/lib/gateway/controllers/curse_of_mirra/user_controller.ex

```diff
@@ -10,6 +10,18 @@ defmodule Gateway.Controllers.CurseOfMirra.UserController do
 
   action_fallback Gateway.Controllers.FallbackController
 
+  def show(conn, %{"id" => user_id}) do
+    game_id = Utils.get_game_id(:curse_of_mirra)
+
+    with {:get_user, {:ok, user}} <- {:get_user, Users.get_user(user_id)},
+         {:curse_user, ^game_id} <- {:curse_user, user.game_id} do
```

Comment:
If we are doing this check then just make a function to fetch by `user_id` and `game_id`, relying on this check is not good

Reply:
Done! I created a `get_user_by_id_and_game_id` function, also i noticed that we're always doing two queries when retrieving users, one to get it and other one to preload some fields, i created this function in a way we only run one query